### PR TITLE
vsr: optimize the order of repair of prepares

### PR DESCRIPTION
**Pull request by matklad**
Instead repairing strictly chronologically, start with commit_min+1, so that we can commit further.

This is helpful in the following scenario:

- the cluster is under high load,
- in particular, the volume of prepares a backup gets through the network overwhelms its disk,
- so a backup doesn't write _all_ incoming prepares to disk
- but still commits them, as they are added to the pipeline_cache

In other words, a replica can end up in a situation where it misses a lot of prepares prior to its own commit min! Before this commit, what would happen is that the replica would repair all those prepares before anything that it actually needs to commit further.

Now, we always try to repair `commit_min+1` first.

### CODE REVIEW

File: src/vsr/replica.zig [hunk 1]
```diff
@@ -7076,26 +7076,45 @@ pub fn ReplicaType(
...
-                        if (budget == 0) {
-                            log.debug("{}: repair_prepares: request budget used", .{self.replica});
-                            return;
+            // Iterate through op_min..=self.op, but make sure to iterate commit_min+1..=self.op
+            // sub range first:
+            // - our first priority is to commit further,
+            // - afterwards, repair commited prepares which are at risk of getting evicted from
+            //   the journal, to help repair any lagging replicas.
+            const repair_ranges_exclusive: [2]struct { u64, u64 } = .{
+                .{ @max(op_min, self.commit_min + 1), self.op + 1 },
```
Comment (chaitanyabhandari) [inline src/vsr/replica.zig:7085]: These checks are so *clean*!

**Review by chaitanyabhandari — APPROVED** [review]
